### PR TITLE
Upgrade submariner to 0.16.3

### PR DESCRIPTION
**Pull request by nirs**
We are using old version, this is not useful for detecting issues early.

### CODE REVIEW

**Comment by nirs** [comment]
Example run:

# Install latest subctl:
```
$ curl -Ls https://get.submariner.io | bash
Installing subctl version latest
  OS detected:           linux
  Architecture detected: amd64
  Download URL:          https://github.com/submariner-io/releases/releases/download/v0.16.3/subctl-v0.16.3-linux-amd64.tar.gz

Downloading...
Download attempt #1...
subctl has been installed as /home/nsoffer/.local/bin/subctl
This provides subctl version: v0.16.3
```

Start env, running submariner self test:
```
$ drenv start envs/submariner.yaml 
2024-02-21 22:00:40,139 INFO    [submariner] Starting environment
2024-02-21 22:00:40,195 INFO    [hub] Starting minikube cluster
2024-02-21 22:00:41,210 INFO    [dr1] Starting minikube cluster
2024-02-21 22:00:42,210 INFO    [dr2] Starting minikube cluster
2024-02-21 22:03:24,255 INFO    [hub] Cluster started in 164.06 seconds
2024-02-21 22:03:24,255 INFO    [hub/0] Running addons/submariner/start
2024-02-21 22:03:47,715 INFO    [dr1] Cluster started in 186.51 seconds
2024-02-21 22:04:25,741 INFO    [dr2] Cluster started in 223.53 seconds
2024-02-21 22:04:51,784 INFO    [hub/0] addons/submariner/start completed in 87.53 seconds
2024-02-21 22:04:51,784 INFO    [hub/0] Running addons/submariner/test
2024-02-21 22:06:10,220 INFO    [hub/0] addons/submariner/test completed in 78.44 seconds
2024-02-21 22:06:10,221 INFO    [submariner] Environment started in 330.08 seconds
```

**Comment by nirs** [comment]
Needs also to fix volsync test.

File: test/addons/submariner/test [hunk 1]
```diff
@@ -128,7 +128,7 @@ def wait_for_service_export(cluster, namespace):
     print(f"Waiting for service in namespace '{namespace}' in cluster '{cluster}'")
     kubectl.wait(
         f"serviceexports/{SERVICE}",
-        "--for=condition=Synced=True",
+        "--for=condition=Ready",
```
Comment (ELENAGER) [inline test/addons/submariner/test:131]: This change is related to the version change?

Comment (nirs) [reply test/addons/submariner/test:131]: Yes, see the linked submariner issue.

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**